ui/app/AppLayouts/Wallet/popups/buy/BuyCryptoModalAdaptor.qml

```diff
@@ -0,0 +1,146 @@
+import QtQml 2.15
+import SortFilterProxyModel 0.2
+
+import StatusQ 0.1
+import StatusQ.Core.Utils 0.1
+
+import utils 1.0
+
+import shared.stores 1.0
+import AppLayouts.Wallet 1.0
+import AppLayouts.Wallet.stores 1.0 as WalletStore
+
+QObject {
```

Comment:
There are two problems from app architecture PoV that we should align and this adaptor seems to be a nice example for that.
The base rule is isolating things and limiting responsibilities of components. Because of that this adaptor should be narrowed:
1. It should not take store and be a proxy for store's actions. Adaptor's responsibility is to transform model data and only that should be taken here. It's ok to explicitly have them in API of UI components.
2. Adaptor should not maintain temporary UI's state (done here via `ModelEntry`). It makes the ui component very dependent on the adaptor.

Ideally UI component should be not dependent on adaptor itself. It should be dependent on data it displays. It should be possible to instantiate UI component in storybook and test in isolation based on mock data only.

It seems to be a bit broader topic and I'm not exhausting the topic here. Let's take it away and discuss a bit more outside gh.